fabric-client-tags-api-v1/src/client/java/net/fabricmc/fabric/api/tag/client/v1/ClientTags.java

```diff
@@ -126,13 +99,28 @@ public static <T> boolean isInWithLocalFallback(TagKey<T> tagKey, RegistryEntry<
 		Optional<? extends Registry<T>> maybeRegistry = getRegistry(tagKey);
 
 		if (maybeRegistry.isPresent()) {
+			// Check the synced tag exists and use that
 			if (maybeRegistry.get().getEntryList(tagKey).isPresent()) {
 				return registryEntry.isIn(tagKey);
 			}
 		}
 
-		if (registryEntry.getKey().isPresent()) {
-			return isInLocal(tagKey, registryEntry.getKey().get());
+		if (registryEntry.getKey().isEmpty()) {
+			// No key?
+			return false;
+		}
+
+		// Recursively search the entries contained with the tag
```

Comment:
Question: Do we need to worry about getting into an infine loop here?

Reply:
I don't think so as tags can't be self-referential, but we could change it.